### PR TITLE
When a user changes their password, 'offer' an option to log out

### PR DESCRIPTION
**Pull request by orta**
It's not really an option, it's pretty mandatory. I'm opening this early, cause I'm not too sure how I can test this. Any ideas?

screen shot 2015-12-09 at 1 21 14 pm

re: https://github.com/artsy/eigen/issues/918

### CODE REVIEW

File: Artsy/Networking/ArtsyAPI.m [hunk 1]
```diff
@@ -126,13 +126,16 @@ - (void)getXappTokenWithCompletion:(void (^)(NSString *xappToken, NSDate *expira
...
-    NSDate *date = [[NSUserDefaults standardUserDefaults] objectForKey:ARXAppTokenExpiryDateDefault];
+    NSDate *xappDate = [[NSUserDefaults standardUserDefaults] objectForKey:ARXAppTokenExpiryDateDefault];
+    NSDate *oauthDate = [[NSUserDefaults standardUserDefaults] objectForKey:AROAuthTokenExpiryDateDefault];
+
+
     NSString *xappToken = [UICKeyChainStore stringForKey:ARXAppTokenKeychainKey];
     NSString *oauthToken = [UICKeyChainStore stringForKey:AROAuthTokenDefault];
 
-    if (date && (xappToken || oauthToken)) {
+    if ((xappDate && xappToken) || (oauthToken && oauthDate)) {
```
Comment (orta) [inline Artsy/Networking/ArtsyAPI.m]: there is a separate date for Oauth, which meant this was _always_ doing networking if you were logged in, even when you didn't need it. Won't be a massive slowdown, but this means less networking.

File: Artsy/View_Controllers/Feed_VCs/ARSimpleShowFeedViewController.m [hunk 2]
```diff
@@ -173,7 +174,11 @@ - (void)refreshFeedItems
             [ARAnalytics finishTimingEvent:ARAnalyticsInitialFeedLoadTime];
 
         } failure:^(NSError *error) {
-            ARErrorLog(@"There was an error getting newest items for the feed: %@", error.localizedDescription);
+            @strongify(self);
+
+            NSHTTPURLResponse *response = error.userInfo[AFURLResponseSerializationErrorDomain]
+                                          ?: error.userInfo[AFNetworkingOperationFailingURLResponseErrorKey];
+            ARErrorLog(@"There was a %@ error getting newest items for the feed: %@", @(response.statusCode), error.localizedDescription);
```
Comment (orta) [inline Artsy/View_Controllers/Feed_VCs/ARSimpleShowFeedViewController.m]: there are multiple places places for the error, so the callback could come one or the other

**Comment by alloy** [comment]
This will have to be updated after merging #983.

**Comment by alloy** [comment]
@orta Is there a reason why we can’t just open the on-boarding views immediately instead of this logout alert and having people restart the app?

**Comment by orta** [comment]
> @orta Is there a reason why we can’t just open the on-boarding views immediately instead of this logout alert and having people restart the app?

generally I feel like it's a lot of extra overhead on the showVC for a pretty niche use case

**Comment by alloy** [comment]
Ok.

Do you want me to merge this one first and you update #983 or will you update this one so that #983 can be merged?

**Comment by orta** [comment]
I'll update one or the other, both are fine :dancer:

**Comment by orta** [comment]
Merged'd #983 into this

**Comment by orta** [comment]
OK, this is :+1: to go

**Pull request merged**